internal/pkg/cli/dialog/use_template_test.go

```diff
@@ -385,8 +385,177 @@ func TestAskUseTemplateOptionsIfNotMet(t *testing.T) {
+						{
+							Id:          "input3",
+							Name:        "input3",
+							Description: "filled in + without default value",
+							Type:        "string",
+							Kind:        "input",
+						},
+					},
+				},
+				{
+					Icon:        "common:settings",
+					Name:        "Step 4",
+					Description: "Step Description",
+					Inputs: []input.Input{
+						{
+							Id:          "input4",
+							Name:        "input4",
+							Description: "filled in + with default value",
+							Type:        "string",
+							Kind:        "input",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Set fake file editor
+	dialog.Prompt.(*interactive.Prompt).SetEditor(`true`)
+
+	// Interaction
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		_, err := console.ExpectString("Select the target branch:")
+		assert.NoError(t, err)
+
+		time.Sleep(20 * time.Millisecond)
+		_, err = console.Send(testhelper.Enter) // enter - Main
+		assert.NoError(t, err)
+
+		_, err = console.ExpectString("Please select which steps you want to fill.")
+		assert.NoError(t, err)
+
+		_, err = console.ExpectString("Select steps:")
+		assert.NoError(t, err)
+
+		time.Sleep(20 * time.Millisecond)
+		_, err = console.Send(DownArrow) // skip step 1
+		assert.NoError(t, err)
+
+		time.Sleep(20 * time.Millisecond)
+		_, err = console.Send(DownArrow) // skip step 2
+		assert.NoError(t, err)
+
+		time.Sleep(20 * time.Millisecond)
+		_, err = console.Send(Space) // select step 3
+		assert.NoError(t, err)
+
+		time.Sleep(20 * time.Millisecond)
+		_, err = console.Send(DownArrow) // move to step 4
+		assert.NoError(t, err)
+
+		time.Sleep(20 * time.Millisecond)
+		_, err = console.Send(Space) // select step 4
+		assert.NoError(t, err)
+
+		time.Sleep(20 * time.Millisecond)
+		_, err = console.Send(Enter) // confirm the selection
+		assert.NoError(t, err)
+
+		_, err = console.ExpectString("Step 3")
+		assert.NoError(t, err)
+
+		_, err = console.ExpectString("input3:")
+		assert.NoError(t, err)
+
+		time.Sleep(20 * time.Millisecond)
+		_, err = console.SendLine("value for input 3")
+		assert.NoError(t, err)
+
+		_, err = console.ExpectString("Step 4")
+		assert.NoError(t, err)
+
+		_, err = console.ExpectString("input4:")
+		assert.NoError(t, err)
+
+		time.Sleep(20 * time.Millisecond)
+		_, err = console.SendLine("value for input 4")
+		assert.NoError(t, err)
+
+		_, err = console.ExpectEOF()
+		assert.NoError(t, err)
+	}()
+
+	output, err := dialog.AskUseTemplateOptions(projectState, stepsGroups, d.Options())
+	assert.NoError(t, err)
+
+	assert.NoError(t, console.Tty().Close())
+	wg.Wait()
+	assert.NoError(t, console.Close())
+
+	// Assert
+	assert.Equal(t, useTemplate.Options{
+		TargetBranch: model.BranchKey{Id: 123},
+		Inputs: template.InputsValues{
+			{Id: "input1", Value: "", Skipped: true},
+			{Id: "input2", Value: "default value", Skipped: true},
+			{Id: "input3", Value: "value for input 3"},
+			{Id: "input4", Value: "value for input 4"},
```

Comment:
A tu je vysledok, prvy `input1` nemal default hodnotu, tak ma prazdnu hodnotu, ... `input2` ma zase default hodnotu.

Pridal som tam aj `Skipped bool`, to sa zatial nikde nepouziva, pouzije sa v dalsom PR.